packages/btcindexer/src/btcindexer.ts

```diff
@@ -326,4 +368,85 @@ export class Indexer {
 		}
 		return updates;
 	}
+	async callSuiMintContract(
+		transaction: Transaction,
+		blockHeight: number,
+		txIndex: number,
+		proof: ProofResult,
+	): Promise<boolean> {
+		try {
+			const tx = new SuiTransactoin();
+			const target = `${this.suiPackageId}::nbtc::mint_nbtc`;
+			tx.moveCall({
+				target: target,
+				arguments: [
+					tx.object(this.suiNbtcObjectId),
+					tx.object(this.suiLightClientObjectId),
+					tx.pure.vector("u8", this.serializeU32(transaction.version)),
+					tx.pure.u32(transaction.ins.length),
+					tx.pure.vector("u8", this.serializeTxInputs(transaction.ins)),
+					tx.pure.u32(transaction.outs.length),
+					tx.pure.vector("u8", this.serializeTxOutputs(transaction.outs)),
+					tx.pure.vector("u8", this.serializeU32(transaction.locktime)),
+					tx.pure.vector(
+						"vector<u8>",
+						proof.proofPath.map((p) => Array.from(p)),
+					),
+					tx.pure.u64(blockHeight),
+					tx.pure.u64(txIndex),
+				],
+			});
+
+			const result = await this.suiClient.signAndExecuteTransaction({
+				signer: this.suiSigner,
+				transaction: tx,
+				options: {
+					showEffects: true,
+				},
+			});
+
+			if (result.effects?.status.status === "success") {
+				console.log(`Mint successful. Digest: ${result.digest}`);
+				return true;
+			} else {
+				console.error(
+					`Mint failed. Digest: ${result.digest}:`,
+					result.effects?.status.error,
+				);
+				return false;
+			}
+		} catch (error) {
+			console.error(`Error during mint call`, error);
+			return false;
+		}
+	}
```

Comment:
how we can test it?

Reply:
Actuall, let's have a separate class:
```
class SuiClient
```

that will have relevant functions. Indexer should take this as an argumetn, instead of the sui config / env in the constructor. This way we can always mock that class and easier test the indexer.

Reply:
Done